apollo-router/src/layers/map_future_with_request_data.rs

```diff
@@ -36,6 +36,7 @@ where
 }
 
 /// [`Service`] for mapping futures with request data. See [`ServiceBuilderExt::map_future_with_request_data()`](crate::layers::ServiceBuilderExt::map_future_with_request_data()).
+#[derive(Clone)]
```

Comment:
Is it required ?

Reply:
Yes, we use it for subgraph service, which should be clonable since it is shared between subgraphs:
https://github.com/apollographql/router/blob/bf66e7ab035be9bcc26a01944710a9199bd3d19e/apollo-router/src/plugins/traffic_shaping/mod.rs#L334